docs/src/app/components/pages/components/TimePicker/ExampleComplex.jsx

```diff
@@ -19,13 +23,15 @@ export default class TimePickerExampleComplex extends React.Component {
           ref="picker12hr"
           format="ampm"
           hintText="12hr Format"
-          onChange={this.handleChangeTimePicker24}
+          value={this.state.value12}
+          onChange={this.handleChangeTimePicker12}
         />
         <TimePicker
           ref="picker24hr"
           format="24hr"
           hintText="24hr Format"
-          onChange={this.handleChangeTimePicker12}
+          value={this.state.value24}
+          onChange={this.handleChangeTimePicker24}
```

Comment:
I removed an early return on  DatePicker#format and that caused this. 
Should be solved once the component starts to make use of utils/DateTime#format